### PR TITLE
Better get ip

### PR DESCRIPTION
**Pull request by armada45-pixel**
Relate to https://github.com/gaurishhs/elysia-ip/issues/24

I think maybe explain in readme like

### How does it work?

For Bun runtime and `headersOnly` is `false`, We use `server.requestIP`.

But if `headersOnly` is `true` or failed to get ip from `server.requestIP`

get ip function will try to get ip from header

getIP function
Priority
- If checkHeaders is string will try to get ip from header in the same name.
- If checkHeaders is empty string like `""` will using default check header list.
- Find ip from header using check header list
- Failed return empty string

Priority list:

1. User specified // Should be remove
2. `X-Forwarded-For` (de-facto standard header)
3. `CF-Connecting-IP` (Cloudflare)
4. `Fastly-Client-IP` (Fastly)
5. `X-Real-IP` (Apache)
6. `X-Client-IP` (Nginx)
7. `X-Cluster-Client-IP` (GCP)
8. `X-Forwarded` (RFC 7239)
9. `Forwarded-For` (RFC 7239)
10. `Forwarded` (RFC 7239)
11. `appengine-user-ip` (GCP)
12. `true-client-ip` (Akamai and Cloudflare)
13. `cf-pseudo-ipv4` (Cloudflare)

### CODE REVIEW

File: src/services/getip.ts [hunk 1]
```diff
@@ -4,36 +4,40 @@ import { logger } from "./logger"
 
 export const getIP = (
   headers: Headers,
-  checkHeaders: IPHeaders[] = headersToCheck,
+  checkHeaders: IPHeaders[] | IPHeaders = headersToCheck,
```
Comment (gaurishhs) [inline src/services/getip.ts:7]: Why can't we just let it remain an Array? What benefit does a string option provide us? Imo we will be slowing down the application this way

Comment (armada45-pixel) [reply src/services/getip.ts:7]: I see on old code, It already support string checkHeaders.
I just add type and improve getIP.
https://github.com/gaurishhs/elysia-ip/blob/bd53e7fc8bb573a91cc3bfa33795d95174c97790/src/services/getip.ts#L9-L23

Comment (gaurishhs) [reply src/services/getip.ts:7]: I get that but it was added by you in prev. PR this seems to be just increasing the complexity

File: src/constants.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ export const headersToCheck: IPHeaders[] = [
   "x-forwarded", // General Forwarded
   "forwarded-for", // RFC 7239
   "forwarded", // RFC 7239
-  "x-forwarded", // RFC 7239
+  "x-forwarded-for", // RFC 7239
```
[the remark this answers is not in the document]
Comment (armada45-pixel) [reply src/constants.ts]: I turn back to original priority in [resort priority](https://github.com/gaurishhs/elysia-ip/pull/25/commits/413e6a05b593c04488a57fdf267f798dcf171814)

**Comment by gaurishhs** [comment]
X-Forwarded-For follows a different format than those headers which is why i had separated it, Having it in `IPHeaders` will break the format [Syntax for X-Forwarded-For](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For#syntax)

I feel we do not need to have a different implementation for just a string header, I want this plugin to be as simple as it can. 

I hope you understand this, I really thank you and appreciate your work on this project.

File: src/services/getip.ts (hunk 1 above)
Comment (armada45-pixel) [reply src/services/getip.ts:7]: https://github.com/gaurishhs/elysia-ip/blob/f46952e0baa39e63aaabe19a2ada291e8ee87036/src/index.ts#L19-L30

**Comment by armada45-pixel** [comment]
Don't forget to change [README.md](https://github.com/gaurishhs/elysia-ip/blob/2188eec2d47ded06f365354096271e57451a56ec/README.md?plain=1#L47-L61) because it conflict with [constants.ts](https://github.com/gaurishhs/elysia-ip/blob/2188eec2d47ded06f365354096271e57451a56ec/src/constants.ts#L3-L16).
https://github.com/gaurishhs/elysia-ip/blob/2188eec2d47ded06f365354096271e57451a56ec/README.md?plain=1#L47-L61
https://github.com/gaurishhs/elysia-ip/blob/2188eec2d47ded06f365354096271e57451a56ec/src/constants.ts#L3-L16

**Comment by gaurishhs** [comment]
Done, thanks for pointing out.

**Pull request closed without merging**